src/System.Management.Automation/engine/Attributes.cs

```diff
@@ -1386,7 +1387,7 @@ public IList<string> ValidValues
         {
             get
             {
-                return _validValues;
+                return _validValues.ToList();
```

Comment:
For the dynamic case - this should probably return a new collection every time it's called.